src/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,178 @@
+	
+	// plugin dependencies
+	private static final String[] dependency = {};
+	
+	//For logging
+	private static final Logger LOG = Logger.getLogger(XSLTInjection.class);
+
+	private static final String[] errorCausingPayloads={"<",">","\'","\""};
+	
+	private static final String[] vendorReturningPayloads={"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/>",
+															"system-property(\'xsl:vendor\')/>",
+															"\"/><xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--",
+															"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--"
+															};
+	private static final String[] xsltVendors={"libxslt","Microsoft","Saxonica","Apache","Xalan","SAXON","Transformiix"};
+	
+	//TODO Add time based payloads (SSRF)
+	
+	@Override
+	public int getId() {
+		return 90017;
+	}
+
+	@Override
+	public String getName() {		
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return dependency;
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {  
+		return technologies.includes(Tech.XML); 
+	}
+	
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.refs");
+	}
+
+	@Override
+	public void init() {
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Initialising XSLT Scanner.");
+		}
+	}
+
+	/*
+	 * This method is called by the active scanner for each GET and POST parameter for every page 
+	 * @see org.parosproxy.paros.core.scanner.AbstractAppParamPlugin#scan(org.parosproxy.paros.network.HttpMessage, java.lang.String, java.lang.String)
+	 */
+	
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+		try {
+			//Count positive Tests
+			int positiveBasedOnError=0; 
+			//error causing tests
+			for (int i=0;i<errorCausingPayloads.length;i++){
+				msg = getNewMsg();
+				setParameter(msg, param, errorCausingPayloads[i]);
+				sendAndReceive(msg);
+				if(!msg.getResponseBody().toString().contains(errorCausingPayloads[i])){
+					positiveBasedOnError++;
+				}
+				// Stop the testing if user requests
+				if (isStop()) {
+					if (LOG.isDebugEnabled()){
+						LOG.debug("Stopping the scan due to a user request");
+					}
+					return;
+				}
+			}
+			//vendor returning tests
+			for(int i=0;i<vendorReturningPayloads.length;i++){
+				msg=getNewMsg();
+				setParameter(msg,param,vendorReturningPayloads[i]);
+				sendAndReceive(msg);
+				for(int j=0;j<xsltVendors.length;j++){
+					if(msg.getResponseBody().toString().contains(xsltVendors[j])){
+							if(positiveBasedOnError>0){						
```

Comment:
This check should be done earlier, before the `vendorReturningPayloads` for loop, no need to try the tests if there's no error (`positiveBasedOnError` will/does not change its value).